share/hashutils.go

```diff
@@ -0,0 +1,11 @@
+package share
+
+import (
+	"crypto/sha256"
+	"hash"
+)
+
+// NewSHA256Hasher returns a new instance of a SHA-256 hasher.
+func NewSHA256Hasher() hash.Hash {
+	return sha256.New()
+}
```

Comment:
It feels a bit overkill to have go file just for this function. Do you mind to move it to share.go?

Reply:
not at all. I moved it to `share.go`!